--- python-authlib.orig/authlib/common/errors.py
+++ python-authlib/authlib/common/errors.py
@@ -43,6 +43,8 @@ class AuthlibHTTPError(AuthlibBaseError)
 
         if self.description:
             error.append(("error_description", self.description))
+        elif isinstance(self.__cause__, AuthlibBaseError):
+            error.append(("error_description", self.__cause__.error))
 
         if self.uri:
             error.append(("error_uri", self.uri))
