--- python-authlib.orig/authlib/oauth2/rfc7523/client.py
+++ python-authlib/authlib/oauth2/rfc7523/client.py
@@ -1,7 +1,7 @@
 import logging
 
 from authlib.jose import jwt
-from authlib.jose.errors import JoseError
+from authlib.jose.errors import JoseError, DecodeError
 
 from ..rfc6749 import InvalidClientError
 
@@ -68,6 +68,8 @@ class JWTBearerClientAssertion:
                 assertion, resolve_key, claims_options=self.create_claims_options()
             )
             claims.validate(leeway=self.leeway)
+        except DecodeError as e:
+            raise InvalidClientError(description="invalid_assertion") from e
         except JoseError as e:
             log.debug("Assertion Error: %r", e)
             raise InvalidClientError(description=e.description) from e
